runtime/frequency/src/lib.rs

```diff
@@ -150,6 +148,32 @@ impl Contains<RuntimeCall> for BaseCallFilter {
 	}
 }
 
+impl BaseCallFilter {
+	fn is_utility_call_allowed(call: &RuntimeCall) -> bool {
+		match call {
+			RuntimeCall::Utility(pallet_utility::Call::batch { calls, .. }) |
+			RuntimeCall::Utility(pallet_utility::Call::batch_all { calls, .. }) |
+			RuntimeCall::Utility(pallet_utility::Call::force_batch { calls, .. }) =>
+				!calls.iter().any(Self::is_batch_call_allowed),
+			_ => false,
+		}
+	}
+
+	fn is_batch_call_allowed(call: &RuntimeCall) -> bool {
+		match call {
+			// Block all `FrequencyTxPayment` calls from utility batch
+			// Block all `Pays::No` calls from utility batch
+			RuntimeCall::FrequencyTxPayment(..) => true,
+			_ if Self::is_pays_no_call(call) => true,
+			_ => false,
+		}
+	}
```

Comment:
@wilwade  @enddynayn  here is updated version

Reply:
Looks good to me